desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -45,40 +50,160 @@ def slack_events(request):
+    # challenge verification
     if slack_message['type'] == 'url_verification':
       response_dict = {"challenge": slack_message['challenge']}
       return JsonResponse(response_dict, status=200)
     
     if 'event' in slack_message:
       event_message = slack_message['event']
       parse_events(event_message)
-  except ValueError as e:
-    raise PopupException(_("Response content is not valid JSON"), detail=e)
-  
+  except ValueError as err:
+    raise PopupException(_("Response content is not valid JSON"), detail=err)
+
   return HttpResponse(status=200)
 
 
-def parse_events(event_message):
-  user_id = event_message.get('user')
-  text = event_message.get('text')
-  channel = event_message.get('channel')
+def parse_events(event):
+  """
+  Parses the event according to its 'type'.
 
-  # ignore bot's own message since that will cause an infinite loop of messages if we respond
-  if event_message.get('bot_id'):
+  """
+  channel_id = event.get('channel')
+  if event.get('type') == 'message':
+    handle_on_message(channel_id, event.get('bot_id'), event.get('text'), event.get('user'))
+
+  if event.get('type') == 'link_shared':
+    handle_on_link_shared(channel_id, event.get('message_ts'), event.get('links'))
+
+
+def handle_on_message(channel_id, bot_id, text, user_id):
+  # ignore bot's own message since that will cause an infinite loop of messages if we respond.
+  if bot_id:
     return HttpResponse(status=200)
   
-  if slack_client is not None:
-    if 'hello hue' in text.lower():
-      response = say_hi_user(channel, user_id)
+  if slack_client:
+    if text and 'hello hue' in text.lower():
+      response = say_hi_user(channel_id, user_id)
+
+      if not response['ok']:
+        raise PopupException(_("Error posting message"), detail=response["error"])
+
+
+def handle_on_link_shared(channel_id, message_ts, links):
+  for item in links:
+    path = urlsplit(item['url'])[2]
+    queryid_or_uuid = urlsplit(item['url'])[3]  # if /hue/editor/ then query_id else if /hue/gist then uuid
+
+    if path == '/hue/editor':
+      query_id = queryid_or_uuid.split('=')[1]
+      doc2 = Document2.objects.get(id=query_id)
+      doc2_data = json.loads(doc2.data)
+
+      statement = doc2_data['snippets'][0]['statement_raw']
+      dialect = doc2_data['dialect'].capitalize()
+      database = doc2_data['snippets'][0]['database'].capitalize()
+      
+      payload = make_query_history_payload(item['url'], statement, dialect, database)
+      response = slack_client.chat_unfurl(channel=channel_id, ts=message_ts, unfurls=payload)
       if response['ok']:
-        return HttpResponse(status=200)
-      else:
-        raise PopupException(response["error"])
+        raise PopupException(_("Cannot unfurl query history link"), detail=response["error"])
 
-  
-def say_hi_user(channel, user_id):
-  """Bot sends Hi<username> message in a specific channel"""
+    if path == '/hue/gist' and ENABLE_GIST_PREVIEW.get():
```

Comment:
Same like this shall we have a flag for /hue/editor too? Or having a single flag or not having at all?

Reply:
:+1: to reuse for now!

Reply:
Okay